services/docker/docker_test.go

```diff
@@ -0,0 +1,102 @@
+package docker
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/honeytrap/honeytrap/pushers"
+)
+
+type Test struct {
+	Name           string
+	ReqMethod      string
+	ReqPath        string
+	ReqBody        string
+	ExpectedBody   string
+	ExpectedStatus int
+}
+
+var tests = []Test{
+	{
+		Name:           "version_root",
+		ReqMethod:      "GET",
+		ReqPath:        "/version",
+		ReqBody:        "",
+		ExpectedStatus: 200,
+		ExpectedBody:   "",
+	},
+	{
+		Name:           "version",
+		ReqMethod:      "GET",
+		ReqPath:        "/1.41/version",
+		ReqBody:        "",
+		ExpectedStatus: 200,
+		ExpectedBody:   "",
+	},
+	{
+		Name:           "info_root",
+		ReqMethod:      "GET",
+		ReqPath:        "/info",
+		ReqBody:        "",
+		ExpectedStatus: 200,
+		ExpectedBody:   "",
+	},
+	{
+		Name:           "info",
+		ReqMethod:      "GET",
+		ReqPath:        "/1.41/info",
+		ReqBody:        "",
+		ExpectedStatus: 200,
+		ExpectedBody:   "",
+	},
+}
+
+func TestDocker(t *testing.T) {
+	//Create Servicer
+	s := Docker()
+
+	// Create channel
+	s.SetChannel(pushers.MustDummy())
+
+	for _, tst := range tests {
+
```

Comment:
you want to wrap all the below in a separate sub tests.

```lang=go
for _, tt := range tests {
    tt.Run(tt.Name, func(t *testing.T) {
          // actual code
    })
```
this way they're executed independently, you get all the errors/timing/etc properly.

Reply:
aaah nice, i was wondering why the output was so rubbish!